core/ecschema-metadata/src/UnitQuery/UnitQuery.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
+import { SchemaContext, SchemaItem, SchemaItemKey, SchemaKey, Unit } from "../ecschema-metadata";
+import { SchemaItemType } from "../ECObjects";
+import { UNIT_DATA } from "./UnitsData";
+
+export class UnitQuery {
```

Comment:
Can you mark this as `alpha` or `beta`?

Also, not the biggest fan of the `UnitQuery` name.  It is not really querying anything but looping over the SchemaContext provided to this specific instance.  If anything, a `UnitsContext` or `UnitProvider` i think would make more sense name wise.

Reply:
Marked as alpha and renamed to UnitProvider